dipy/segment/tissue.py

```diff
@@ -81,51 +82,36 @@ def classify(self, image, nclasses, beta, tolerance=None, max_iter=None):
         final_segmentation = np.empty_like(image)
         initial_segmentation = seg_init
 
-        allow_break = max_iter is None or tolerance is not None
-
-        if max_iter is None:
-            max_iter = 100
-
-        if tolerance is None:
-            tolerance = 1e-05
-
         for i in range(max_iter):
-
             if self.verbose:
-                print('>> Iteration: ' + str(i))
+                print(f'>> Iteration: {i}')
 
             PLN = icm.prob_neighborhood(seg_init, beta, nclasses)
             PVE = com.prob_image(image_gauss, nclasses, mu, sigmasq, PLN)
 
-            mu_upd, sigmasq_upd = com.update_param(image_gauss,
-                                                   PVE, mu, nclasses)
+            mu_upd, sigmasq_upd = com.update_param(image_gauss, PVE, mu,
+                                                   nclasses)
             ind = np.argsort(mu_upd)
             mu_upd = mu_upd[ind]
             sigmasq_upd = sigmasq_upd[ind]
 
-            negll = com.negloglikelihood(image_gauss,
-                                         mu_upd, sigmasq_upd, nclasses)
-            final_segmentation, energy = icm.icm_ising(negll,
-                                                       beta, seg_init)
+            negll = com.negloglikelihood(image_gauss, mu_upd, sigmasq_upd,
+                                         nclasses)
+            final_segmentation, energy = icm.icm_ising(negll, beta, seg_init)
 
-            if allow_break:
-                energy_sum.append(energy[energy > -np.inf].sum())
+            energy_sum.append(energy[energy > -np.inf].sum())
 
             if self.save_history:
                 self.segmentations.append(final_segmentation)
                 self.pves.append(PVE)
                 self.energies.append(energy)
-                if allow_break:
-                    self.energies_sum.append(energy_sum[-1])
-                else:
-                    self.energies_sum.append(energy[energy > -np.inf].sum())
-
-            if allow_break and i > 5:
+                self.energies_sum.append(energy_sum[-1])
 
+            if tolerance > 0 and i > 5:
```

Comment:
For a separate PR: if 5 means something here, it should be put to a variable and explain the meaning, as it is also used in https://github.com/dipy/dipy/pull/3132/files#diff-540e9ecc7f5c0befbaff5a9def01d422445705275835db8a00e8c763792e6e12R114.

Reply:
Agree. We could give it a name like tolerance_window

Reply:
I can put up a separate PR for that. Thank you for the feedback.